catroid/src/main/java/org/catrobat/catroid/formulaeditor/InternToExternGenerator.java

```diff
@@ -319,6 +308,26 @@ private String generateExternStringFromToken(InternToken internToken, boolean tr
 		}
 	}
 
+	private String generateExternNumberStringForToken(String number, boolean trimNumbers) {
+
+		if (!number.contains(".")) {
+			return number;
+		}
+
+		if (trimNumbers) {
+			return cutTrailingZerosofString(number);
+		}
+
+		String left = number.substring(0, number.indexOf('.'));
+		String right = number.substring(number.indexOf('.') + 1);
+
+		return left + getExternStringForInternTokenValue(".", context) + right;
+	}
+
+	private String cutTrailingZerosofString(String number) {
+		return new BigDecimal(number).stripTrailingZeros().toPlainString();
```

Comment:
It may take a while to approve this PR since numbers are always fully written out now. e.g. `0.00001` will be written as `0.00001`, previously `1.0E-5`. I've added a comment [here](https://jira.catrob.at/browse/CAT-1295).

Reply:
hey,
i've analyzed where this behaviour came from in the old version and Float.parseFloat does that trick. I could not find a solution doing this in the string representation itself so I had to call this function.
I've made a new commit for now so I could easily go back to my previous version 
Now:
0,0001 = 1.0E-4

Would be nice if you could have a quick look sometime, so I know if I should remove that commit or ammend it :)